### ISSUE TITLE
[Bug] RayService example in docs generates an error

### ISSUE THREAD

**Issue by smit-kiri** [ray-project/kuberay#977]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/kuberay/issues) and found no similar issues.

### KubeRay Component

ray-operator

### What happened + What you expected to happen

When querying the `RayService` fruit stand example described in the [Kubernetes production guide](https://docs.ray.io/en/latest/serve/production-guide/kubernetes.html#deploying-a-serve-application), I see this error:

```
curl -X POST -H 'Content-Type: application/json' localhost:8000 -d '["MANGO", 2]'
Task Error. Traceback: ray::ServeReplica:DAGDriver.handle_request() (pid=XXX, ip=XX.X.XXX.XXX)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/serve/_private/utils.py", line 231, in wrap_to_ray_error
    raise exception
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/serve/_private/replica.py", line 420, in invoke_single
    result = await method_to_call(*args, **kwargs)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/serve/drivers.py", line 132, in __call__
    await self.app(request.scope, receive=request.receive, send=sender)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/fastapi/applications.py", line 269, in __call__
    await super().__call__(scope, receive, send)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/starlette/applications.py", line 124, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/home/ray/anaconda3/lib/python3.7/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/starlette/exceptions.py", line 93, in __call__
    raise exc
  File "/home/ray/anaconda3/lib/python3.7/site-packages/starlette/exceptions.py", line 82, in __call__
    await self.app(scope, receive, sender)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
    raise e
  File "/home/ray/anaconda3/lib/python3.7/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/starlette/routing.py", line 670, in __call__
    await route.handle(scope, receive, send)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/starlette/routing.py", line 266, in handle
    await self.app(scope, receive, send)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/starlette/routing.py", line 65, in app
    response = await func(request)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/fastapi/routing.py", line 228, in app
    dependant=dependant, values=values, is_coroutine=is_coroutine
  File "/home/ray/anaconda3/lib/python3.7/site-packages/fastapi/routing.py", line 160, in run_endpoint_function
    return await dependant.call(**values)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/serve/drivers.py", line 126, in handle_request
    return await self.predict(inp)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/serve/drivers.py", line 138, in predict
    await self.dags[self.MATCH_ALL_ROUTE_PREFIX].remote(*args, **kwargs)
ray.exceptions.RayTaskError(ValueError): ray::ServeReplica:FruitMarket.handle_request() (pid=144, ip=10.4.172.39)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/serve/_private/utils.py", line 231, in wrap_to_ray_error
    raise exception
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/serve/_private/replica.py", line 420, in invoke_single
    result = await method_to_call(*args, **kwargs)
  File "/tmp/ray/session_2022-11-06_16-02-47_744657_24/runtime_resources/working_dir_files/https_github_com_ray-project_test_dag_archive_c620251044717ace0a4c19d766d43c5099af8a77/fruit.py", line 32, in check_price
    return ray.get(fruit_stand.check_price.remote(amount))
ValueError: 'object_refs' must either be an object ref or a list of object refs..%
```

As described in [this slack thread](https://ray-distributed.slack.com/archives/C02GFQ82JPM/p1667782377713769), I was able to resolve this by setting `SERVE_DEPLOYMENT_HANDLE_IS_SYNC` to `"1"`

### Reproduction script

Follow the documentation [here](https://docs.ray.io/en/latest/serve/production-guide/kubernetes.html#deploying-on-kubernetes). To install the KubeRay operator, follow [this guide](https://ray-project.github.io/kuberay/deploy/installation/#single-namespace-version) to install the single namespace version. You will see the error when querying the application.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by kevin85421** [comment]
The document will be fixed after Ray 2.4.0 is officially out.

**Comment by kevin85421** [comment]
Fix in https://github.com/ray-project/ray/pull/34178
